sycl/source/detail/queue_impl.hpp

```diff
@@ -118,6 +118,9 @@ class queue_impl {
   }
 
   ~queue_impl() {
+    if (has_property<property::queue::wait_on_destruction>()) {
```

Comment:
It doesn't seem like enough changes since queue is reference counted and can be held by other non user visible entities.
Could you please provide an example(test) demonstrating when wait is supposed to be called with respect to a user application.

Reply:
Where? I'm confused what we do about tests now since we moved them all out of tree.  It was very simple when the tests for the runtime were with the runtime.

Reply:
> It was very simple when the tests for the runtime were with the runtime.
+1 

The tests can be added there: https://github.com/intel/llvm-test-suite